service/history/api/updateworkflow/api.go

```diff
@@ -162,6 +163,8 @@ func addWorkflowTaskToMatching(
 		ScheduledEventId:       task.ScheduledEventID,
 		ScheduleToStartTimeout: taskScheduleToStartTimeout,
 		Clock:                  clock,
+		// TODO: is it ever possible that this is the first workflow task?
+		IsFirstWorkflowTask: ms.GetLastWorkflowTaskStartedEventID() == common.EmptyEventID,
```

Comment:
I'd need to read the code to answer this but intuitively it seems like this condition is accurate.
I'd get a second opinion from people with more milage in this codebase.